File: js/binary_tree.js
function Node(data){
	this.data = data;
	this.left = null;
	this.right = null;
};

function height(root){
	if(root == null)
		return 0;

	var lHeight = height(root.left);
	var rHeight = height(root.right);

	return Math.max(lHeight, rHeight) + 1;
}

function buildLinkedListWithBFS(){
	var root = new Node(1);
	root.left = new Node(2);
	root.right = new Node(3);
	root.left.left = new Node(4);
	root.left.right = new Node(5);
	root.right.left = new Node(6);
	root.right.right = new Node(7);

	function BFS(){
		var res = [];
		var curr = [];

		curr.push(root);

		while(curr.length){
			res.push(curr);
			var parent = Object.assign(curr);
			curr = [];
			for(var i=0; i<parent.length; i++){
				if(parent[i].left){
					curr.push(parent[i].left);
				}
				if(parent[i].right){
					curr.push(parent[i].right);
				}
			}
		}

		return res;
	}

	console.log(BFS())
}


var root = new Node(56);
root.left = new Node(1);
root.right = new Node(2);
root.left.left = new Node(3);
root.left.left.left = new Node(4);


//console.log(height(root));

buildLinkedListWithBFS();

File: java/BinaryTree.java
public class BinaryTree
{
	public Node root;
	public class Node
	{
		int data;
		Node left;
		Node right;

		public Node(int d)
		{
			data = d;
			left = null;
			right = null;
		}
	}

	public int height(Node root){
		if(root == null) {
			return 0;
		}

		int lh = height(root.left);
		int rh = height(root.right);

		return 1 + Math.max(lh, rh);
	}

	public int diameter(Node root){
		if(root == null) {
			return 0;
		}

		int lh = height(root.left);
		int rh = height(root.right);

		mainDiameter = 1 + lh + rh;



	}

		
	

	public static void main(String[] args)
	{
		BinaryTree bt = new BinaryTree();

		bt.root =  bt.new Node(5);
		bt.root.left = bt.new Node(1);
		bt.root.right = bt.new Node(4);
		bt.root.left.left = bt.new Node(3);
		bt.root.left.right = bt.new Node(7);
		bt.root.left.left.left = bt.new Node(2);

		System.out.println(bt.height(bt.root));
		System.out.println("aaadsfa");
	}
}